pkg/controllers/vdb/upgrade.go

```diff
@@ -87,15 +90,22 @@ func (i *UpgradeManager) IsUpgradeNeeded(ctx context.Context, sandbox string) (b
 
 // isUpgradeInProgress returns true if state indicates that an upgrade
 // is already occurring.
-func (i *UpgradeManager) isUpgradeInProgress() bool {
+func (i *UpgradeManager) isUpgradeInProgress(sbName string) bool {
 	// We first check if the status condition indicates the upgrade is in progress
-	isSet := i.Vdb.IsStatusConditionTrue(i.StatusCondition)
+	isSet := i.IsUpgradeStatusTrue(sbName)
 	if isSet {
 		i.ContinuingUpgrade = true
 	}
 	return isSet
 }
 
+func (i *UpgradeManager) IsUpgradeStatusTrue(sbName string) bool {
```

Comment:
Can we avoid exporting if we can?
```suggestion
func (i *UpgradeManager) isUpgradeStatusTrue(sbName string) bool {
```